### PR TITLE
Task management resource

### PR DESCRIPTION
**Pull request by olive42**
This lets you create scheduled tasks on the Nexus server.

### CODE REVIEW

File: resources/task.rb [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  def apiclient
+    @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
+  end
+
+  begin
+    res = apiclient.run_script('get_task', desired.task_name)
+    current_value_does_not_exist! if res == 'null'
+    config = JSON.parse(res)
+    ::Chef::Log.warn "Config is: #{config}"
+    task_name config['.name']
```
Comment (Annih) [inline resources/task.rb]: This is an identity property, so there is no need to set it. And could you explain me why there is a *.* before `name` ?

File: resources/task.rb [hunk 2]
```diff
@@ -0,0 +1,149 @@
+property :task_name, String, desired_state: false, identity: true, name_attribute: true
+property :task_source, String, desired_state: false, identity: true, default: ''.freeze
```
Comment (Annih) [inline resources/task.rb]: This is a "desired_state" property and not an "identity" one.

File: resources/task.rb [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+property :task_crontab, String, desired_state: false, identity: true, default: '0 1 * * * ?'.freeze
```
Comment (Annih) [inline resources/task.rb]: This is a "desired_state" property and not an "identity" one

File: resources/task.rb [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                                default: lazy { node['nexus3']['api']['password'] }
+
+load_current_value do |desired|
+  api_endpoint desired.api_endpoint
+  api_username desired.api_username
+  api_password desired.api_password
+
+  def apiclient
+    @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
+  end
```
Comment (Annih) [inline resources/task.rb]: instead of doing a local function, please use a variable

File: resources/task.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                                default: lazy { node['nexus3']['api']['endpoint'] }
+property :api_username, String, desired_state: false, identity: true,
+                                default: lazy { node['nexus3']['api']['username'] }
+property :api_password, String, desired_state: false, identity: true, sensitive: true,
+                                default: lazy { node['nexus3']['api']['password'] }
+
+load_current_value do |desired|
+  api_endpoint desired.api_endpoint
+  api_username desired.api_username
+  api_password desired.api_password
```
Comment (Annih) [inline resources/task.rb]: The 3 above lines are useless => identity properties

File: resources/task.rb [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+  def apiclient
+    @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
+  end
+
+  begin
+    res = apiclient.run_script('get_task', desired.task_name)
+    current_value_does_not_exist! if res == 'null'
+    config = JSON.parse(res)
```
Comment (Annih) [inline resources/task.rb]: Rewrite the 3 above lines:
```ruby
config = ::JSON.parse(apiclient_run_script('get_task', task_name)
current_value_does_not_exist! if res.nil?
```

File: resources/task.rb [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+      args name: new_resource.task_name,
+           source: new_resource.task_source,
+           crontab: new_resource.task_crontab
+
+      action %i(create run)
+      endpoint new_resource.api_endpoint
+      username new_resource.api_username
+      password new_resource.api_password
+
+      content <<-EOS
```
Comment (Annih) [inline resources/task.rb]: Move the content to the file directory and use `::Nexus3::Scripts.groovy_content('upsert_task', node)`

File: resources/task.rb [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+action :delete do
+  init
+
+  converge_if_changed do
+    nexus3_api "delete_task #{new_resource.task_name}" do
+      action %i(create run)
+      script_name 'delete_task'
+      args new_resource.task_name
+
+      content <<-EOS
```
Comment (Annih) [inline resources/task.rb]: Move the content to the file directory and use `::Nexus3::Scripts.groovy_content('delete_task', node)`

File: resources/task.rb [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+    nexus3_api "get_task #{new_resource.task_name}" do
+      action :create
+      script_name 'get_task'
+      args new_resource.task_name
+      endpoint new_resource.api_endpoint
+      username new_resource.api_username
+      password new_resource.api_password
+
+      content <<-EOS
```
Comment (Annih) [inline resources/task.rb]: Move the content to the file directory and use `::Nexus3::Scripts.groovy_content('get_task', node)`

File: resources/task.rb (hunk 1 above)
Comment (olive42) [reply resources/task.rb]: removed.

File: resources/task.rb (hunk 6 above)
Comment (olive42) [reply resources/task.rb]: done

File: resources/task.rb (hunk 5 above)
Comment (olive42) [reply resources/task.rb]: done

File: resources/task.rb (hunk 4 above)
Comment (olive42) [reply resources/task.rb]: done

File: resources/task.rb (hunk 3 above)
Comment (olive42) [reply resources/task.rb]: done

File: resources/task.rb (hunk 2 above)
Comment (olive42) [reply resources/task.rb]: done

File: resources/task.rb (hunk 7 above)
Comment (olive42) [reply resources/task.rb]: done

File: resources/task.rb (hunk 8 above)
Comment (olive42) [reply resources/task.rb]: done

File: resources/task.rb (hunk 9 above)
Comment (olive42) [reply resources/task.rb]: done

**Pull request merged**